### PR TITLE
Use query in cardinality check

### PR DESCRIPTION
**Pull request by benwtrent**
When checking the cardinality of a field, the query should be take into account. The user might know about some bad data in their index and want to filter down to the target_field values they care about.

### CODE REVIEW

**Comment by benwtrent** [comment]
run elasticsearch-ci/2

File: x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/MlNativeDataFrameAnalyticsIntegTestCase.java [hunk 1]
```diff
@@ -170,6 +172,17 @@ protected static DataFrameAnalyticsConfig buildAnalytics(String id, String sourc
             .build();
     }
 
+    protected static DataFrameAnalyticsConfig buildAnalytics(String id, String sourceIndex, String destIndex,
+                                                             @Nullable String resultsField, DataFrameAnalysis analysis,
+                                                             QueryBuilder queryBuilder) throws Exception {
+        return new DataFrameAnalyticsConfig.Builder()
```
Comment (przemekwitek) [inline x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/MlNativeDataFrameAnalyticsIntegTestCase.java]: The other method called `buildAnalytics` (starting in line 165) should call this one, as the new method is more generic.

**Review by dimitris-athanasiou — APPROVED** [review]
LGTM Nice one!

File: x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/ClassificationIT.java [hunk 2]
```diff
@@ -248,6 +250,27 @@ public void testDependentVariableCardinalityTooHighError() {
         assertThat(e.getMessage(), equalTo("Field [keyword-field] must have at most [2] distinct values but there were at least [3]"));
     }
 
+    public void testDependentVariableCardinalityTooHighErrorButWithQuery() throws Exception {
+        initialize("cardinality_too_high");
```
Comment (przemekwitek) [inline x-pack/plugin/ml/qa/native-multi-node-tests/src/test/java/org/elasticsearch/xpack/ml/integration/ClassificationIT.java]: Please make the names assigned to jobs unique (currently this one and the one in line 236 are the same).

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Review by przemekwitek — APPROVED** [review]
LGTM

**Pull request merged**